### DISCUSSION TITLE
We are programmers, not tiktokers

### DISCUSSION THREAD

**Discussion by TheOtterlord** [community/community#13169] [Other Feature Feedback, Questions, & Ideas]
*Ok, so this turned out to be a bit of a ramble, but I think the points made here are important to this feature being useful (especially after reviewing feedback from other devs).*

I'm honestly not sure how I feel about the "For You" feed, but I just hope this isn't a path of "social mediafying" the platform. Either way, this could turn out to be a useful feature or just problematic.
On one hand, it has potential, mostly for people who like to contribute to open-source projects, or try new things. But the "Explore" page (and other sites) has always done this reasonably well.
On the other hand, the "For you" page at least for me is filled with nothing but noise. People who contributed to similar projects to me, or who reported issues, or contributed once to my projects (or vice-versa) take up the majority of the feed. And most of those are echoing which repositories they have starred, which is irrelavent to our interactions, and not related to anything I use (except by programming languages sometimes).

## Social Media Metrics

GitHub is not a social media platform, so using similar metrics to other platforms (such as likes/stars, follows, etc) don't give any form of relevant feed for the majority of developers. While the "For You" page could be awesome for discovering new projects and developers, I think different metrics might be better for choosing these cards.
Seeing when people you follow and contribute towards (or viceversa) create new projects, or maybe even clone existing projects could be cool (although see my thoughts on configuration below), but the number of repos people star end up clogging up the (potentially) more awesome stuff and make the page feel like a bland activity log of people who you have talked with in issues or PRs.

## Configuration

As this feature is likely to be added (given there are plenty of people who do agree with this idea, although with their own opinions on potential improvements) it would be awesome to at least have a decent way of personalising this ourselves. I think this would have to include an option to use the old activity feed, but for those who wish to use it, atleast having the ability to choose which types of activity cards are displayed would be the most simple configuration that would make a massive difference. In addition, the ability to control who I see in the feed (followed, followers, people I've interacted with recently/all time, etc) would also make this a potentially more useful feature.

**Comment by solarkraft** [comment]
> but I just hope this isn't a path of "social mediafying" the platform

IMO this has been always GitHub's core. Collaboration is good, but the core reason I have projects on GitHub is discoverability and generally social things. As a user who likes to discover cool new stuff I thus see a lot of potential. It also allowed me to see a release of a project I care about that I probably wouldn't have known about otherwise.

**Reply by TheOtterlord** [reply]
That's fair. Though discoverability through things like explore has never been a big thing. Occationally you might get a few people from there, but I don't think that's how the vast majority of people discover projects. And I see the same thing happening with this FYP.

As for social aspects of open-source, most projects use Discord, Matrix, Reddit, and other social sites that are built to handle these interactions. Don't get me wrong, I think features like GitHub Discussions are great, but they're not suitable for every topic or discussion format.

Yeah, seeing releases of projects is nice, although not all of them are relevant (as in I haven't starred them, used their project anyway, nor even heard of them at all) the ones that are I find make for a useful information feed. And this is the sort of thing I'd like to see on my feed (hint to request for customisation😉)

**Comment by hacknorris-code** [comment]
hope this tab is only a joke and will be removed soon. i also feel it. would we maybe work on an addon to block it in chrome and firefox? (i think on mobile it can be hard to do without root)

**Reply by hacknorris-code** [reply]
+ i see - im closer to move fully to codeberg :/

**Reply by TheOtterlord** [reply]
I'm not sure an addon to remove it would be needed. All GitHub needs to do is acknowledge that this feature is not for everyone. I don't see the old activity tab going anywhere, but just the ability to hide the page would be nice (it's in beta now, so this might already be planned for the actual release). And I think that with some configuration options, the majority of people would be able to find some use in it (though still having the old tab would be nice for those who really don't want a FYP)

**Reply by hacknorris-code** [reply]
i know right? i already inspected this feed button element tbh:

```
  <button data-hydro-click="{&quot;event_type&quot;:&quot;feeds.feed_click&quot;,&quot;payload&quot;:{&quot;click_target&quot;:&quot;feed.next_tab&quot;,&quot;originating_url&quot;:&quot;https://github.com/&quot;,&quot;user_id&quot;:97974194}}" data-hydro-click-hmac="58db70ada08297bb870906a7f457b3b00f822be838ad4ac8c72926ee15b36bfd" id="feed-next" type="button" role="tab" aria-controls="panel-2" data-view-component="true" class="UnderlineNav-item hx_underlinenav-item-ts lh-0">
    
      <span data-view-component="true">                For you
                <span title="Feature Release Label: Beta" aria-label="Feature Release Label: Beta" data-view-component="true" class="Label Label--success Label--inline ml-1 px-1">Beta</span>
</span>
    
</button>
```

**Reply by hacknorris-code** [reply]
+ i dont think github will make it disablable. i think they'll go more liike twitter does (almost forces personalized feed by creating crashes resetting cookies)

**Comment by jamieshark** [comment]
Customizing the feed / being able to create "sub-feeds" is an idea that has come up. Ideally, we'd be able to give enough control to the user to filter and sort content that feels relevant to them (or a current project / interest / etc.) and display the information in a way that is easy to consume. I'm curious to know (aside from the things you've mentioned already) what sort of "configuration" settings would make the feed most useful and/or usable for you?

**Reply by TheOtterlord** [reply]
I think once you are able to customise who you see, and what you see (stars, releases, etc) you control most of what you'd be able to reasonably configure. Perhaps some input on topics and languages (or just whether or not your topic stars play a part) for those who prefer certain languages, or have topics they'd like to see more from.

**Reply by jamieshark** [reply]
Thanks!

**Comment by stretch07** [comment]
Github's goal was to be 'social coding'. While this is not exactly a form of social media, I wouldn't be surprised if this were here to stay. Personally, however, I use Github to meet new programmers and discover new projects. This 'for you' tab is kind of a dream come true for me. Github should probably listen if there is a huge demand to remove it.

**Comment by Xetera** [comment]
The only resemblance I can see between this feature and tiktok is the name of the tab. Github could've cranked the gamification wayyy up like tiktok with how they reward discovery but they didn't. This is just a different way of discovering new things that's like the explore page but slightly better in my experience. I just don't see what harm the new tab brings, this is a social platform.

**Reply by TheOtterlord** [reply]
A social platform of sorts. It is primarily a collaboration platform. I don't mind new features like this, but I do think there are ways to improve it, and I also see why people don't want this as their feed at all, and so I think it should be an option considering the volume of people voicing concerns. And yeah, this isn't overgamification yet (luckily, as I don't want this to end up like SO because I really like github).
(ps: yeah the title came from a discord group chat xD)

**Comment by ghost** [comment]
I just want to be able to remove this tab or hide a certain people from my feed. Thank you.

**Comment by jfullstackdev** [comment]
hey guys, there is a new update on the For You feed, you can now filter it, and for those who don't want it can simply disable it by unchecking all filters

![image](https://user-images.githubusercontent.com/47092464/182012117-0b19be9d-e849-462e-909e-d20d28915c1d.png)
